test/functorch/test_aotdispatch.py

```diff
@@ -5291,6 +5291,25 @@ def f(a, b):
         self.assertEqual(a_ref_base.grad.a, a_test_base.grad.a)
         self.assertEqual(a_ref_base.grad.b, a_test_base.grad.b)
 
+    def test_aot_dispatch_output_requires_grad_in_no_grad(self):
+        def fn(x):
+            with torch.enable_grad():
+                out = x + 1
+            return out
+
+        inp_fns = [
+            lambda: torch.ones(10, requires_grad=True),
+            lambda: torch.ones(10, requires_grad=False),
+        ]
+
+        compiled_f = aot_function(fn, nop)
+        for inp_fn in inp_fns:
+            with torch.no_grad():
+                ref_out = fn(inp_fn())
+                out = compiled_f(inp_fn())
+
+                self.assertEqual(ref_out.requires_grad, out.requires_grad)
```

Comment:
nit: can you also test that our outputs + gradients are actually correct? A slightly more complicated test might be better:
```
def fn(x):
     out1 = x.sin()
     with torch.enable_grad():
         out2 = x.cos()
    return out1, out2

x = torch.randn(10, requires_grad=True)
with torch.no_grad():
    out1, out2 = f(x)
(out1 + out2).sum().backward()
# check that x.grad is the same compared to when we use `aot_function`
```